extensions/ql-vscode/src/common/vscode/commands.ts

```diff
@@ -40,13 +40,13 @@ export function createVSCodeCommandManager<
  * @param logger The logger to use for error reporting.
  * @param telemetry The telemetry listener to use for error reporting.
  */
-export function registerCommandWithErrorHandling(
+export function registerCommandWithErrorHandling<S extends unknown[], T>(
```

Comment:
It's annoying that `registerCommandWithErrorHandling` gained two args, but thankfully in all the cases we use it the types are inferred so we don't need to do anything. In reality in both uses the type could be `() => void`, so I wouldn't be against just hardcoding that here for simplicity, and we can revisit later if this function needs to handle more complex functions.

Reply:
I'm happy either way

Reply:
I've had another go at this and condensed the two type args into one. I don't know why I didn't do this before. Maybe it didn't work before and some other change made it possible, or maybe I just didn't do it properly.

I think this is a better solution overall and makes things clearer.